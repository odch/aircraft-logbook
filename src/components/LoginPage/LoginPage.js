import React from 'react'
import PropTypes from 'prop-types'
import { Redirect } from 'react-router-dom'
import { FormattedMessage } from 'react-intl'
import Button from '@material-ui/core/Button'
import withStyles from '@material-ui/core/styles/withStyles'
import LoginForm from '../../containers/LoginFormContainer'

const styles = theme => ({
  layout: {
    width: 'auto',
    display: 'block',
    marginTop: theme.spacing.unit * 8,
    marginLeft: theme.spacing.unit * 3,
    marginRight: theme.spacing.unit * 3,
    [theme.breakpoints.up(400 + theme.spacing.unit * 3 * 2)]: {
      width: 400,
      marginLeft: 'auto',
      marginRight: 'auto'
    }
  },
  registrationButton: {
    marginTop: '1em'
  }
})

class LoginPage extends React.Component {
  render() {
    if (!this.props.auth.isEmpty) {
      return <Redirect to="/" />
    }

    const { classes } = this.props

    return (
      <main className={classes.layout}>
        <LoginForm />
        <Button
          className={classes.registrationButton}
          href="/register"
          variant="raised"
          color="primary"
          data-cy="registration"
          fullWidth
        >
          <FormattedMessage id="login.registration" />
        </Button>
      </main>
    )
  }
}

LoginPage.propTypes = {
  auth: PropTypes.shape({
    isEmpty: PropTypes.bool.isRequired
  }),
  classes: PropTypes.object.isRequired
}

export default withStyles(styles)(LoginPage)
