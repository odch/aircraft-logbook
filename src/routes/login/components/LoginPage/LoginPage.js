import React from 'react'
import PropTypes from 'prop-types'
import { Redirect } from 'react-router-dom'
import { FormattedMessage } from 'react-intl'
import Button from '@material-ui/core/Button'
import withStyles from '@material-ui/core/styles/withStyles'
import Typography from '@material-ui/core/Typography'
import featureToggles from 'feature-toggles'
import GoogleLogin from '../../containers/GoogleLoginContainer'
import LoginForm from '../../containers/LoginFormContainer'
import getAuthQueryToken from '../../../../util/getAuthQueryToken'
import LoadingIcon from '../../../../components/LoadingIcon'

const styles = theme => ({
  layout: {
    width: 'auto',
    display: 'block',
    marginTop: theme.spacing(8),
    marginLeft: theme.spacing(3),
    marginRight: theme.spacing(3),
    [theme.breakpoints.up(400 + theme.spacing(3 * 2))]: {
      width: 400,
      marginLeft: 'auto',
      marginRight: 'auto'
    }
  },
  registrationButton: {
    marginTop: '1em'
  },
  google: {
    marginBottom: '2em'
  }
})

class LoginPage extends React.Component {
  componentDidMount() {
    const { auth, router, loginWithToken } = this.props
    if (auth.isEmpty) {
      const queryToken = getAuthQueryToken(router.location)
      if (queryToken) {
        loginWithToken(queryToken)
      }
    }
  }

  render() {
    const { auth, router, tokenLogin, classes } = this.props

    if (!auth.isEmpty) {
      if (router.location.state && router.location.state.from) {
        return <Redirect to={router.location.state.from.pathname} />
      }
      return <Redirect to="/" />
    }

    if (tokenLogin.submitted) {
      return <LoadingIcon />
    }

    if (tokenLogin.failed) {
      return (
        <main className={classes.layout}>
          <FormattedMessage id="login.tokeninvalid" />
        </main>
      )
    }

    return (
      <main className={classes.layout}>
        <GoogleLogin className={classes.google} />
        {featureToggles.isFeatureEnabled('emailPasswordAuth') && (
          <React.Fragment>
            <Typography align="center" color="textSecondary" gutterBottom>
              <FormattedMessage id="login.or" />
            </Typography>
            <LoginForm />
            {featureToggles.isFeatureEnabled('registration') && (
              <Button
                className={classes.registrationButton}
                href="/register"
                variant="text"
                color="primary"
                data-cy="registration"
                fullWidth
              >
                <FormattedMessage id="login.registration" />
              </Button>
            )}
          </React.Fragment>
        )}
      </main>
    )
  }
}

LoginPage.propTypes = {
  auth: PropTypes.shape({
    isEmpty: PropTypes.bool.isRequired
  }).isRequired,
  classes: PropTypes.object.isRequired,
  router: PropTypes.shape({
    location: PropTypes.shape({
      search: PropTypes.string,
      state: PropTypes.shape({
        queryToken: PropTypes.shape({
          orgId: PropTypes.string.isRequired,
          token: PropTypes.string.isRequired
        }),
        from: PropTypes.shape({
          pathname: PropTypes.string.isRequired
        })
      })
    }).isRequired
  }).isRequired,
  tokenLogin: PropTypes.shape({
    submitted: PropTypes.bool.isRequired,
    failed: PropTypes.bool.isRequired
  }).isRequired,
  loginWithToken: PropTypes.func.isRequired
}

export default withStyles(styles)(LoginPage)
