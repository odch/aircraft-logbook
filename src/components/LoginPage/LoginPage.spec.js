import React from 'react'
import ShallowRenderer from 'react-test-renderer/shallow'
import LoginPage from './LoginPage'

describe('components', () => {
  describe('LoginPage', () => {
    it('redirects to start page if authenticated', () => {
      const auth = {
        isEmpty: false
      }
      const tree = new ShallowRenderer().render(<LoginPage auth={auth} />)
      expect(tree).toMatchSnapshot()
    })

    it('renders login form if not authenticated', () => {
      const auth = {
        isEmpty: true
      }
      const tree = new ShallowRenderer().render(<LoginPage auth={auth} />)
      expect(tree).toMatchSnapshot()
    })
  })
})
