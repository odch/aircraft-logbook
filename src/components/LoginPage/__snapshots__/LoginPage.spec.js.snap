// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components LoginPage redirects to start page if authenticated 1`] = `
<div>
  start page
</div>
`;

exports[`components LoginPage renders login form if not authenticated 1`] = `
<main
  className="LoginPage-layout-1"
>
  <form
    onSubmit={[Function]}
  >
    <div
      className="MuiFormControl-root-4 MuiFormControl-marginNormal-5 MuiFormControl-fullWidth-7"
    >
      <label
        className="MuiFormLabel-root-15 MuiFormLabel-filled-19 MuiFormLabel-required-20 MuiInputLabel-root-8 MuiInputLabel-formControl-9 MuiInputLabel-animated-12 MuiInputLabel-shrink-11"
        data-shrink={true}
        htmlFor="email"
      >
        <span>
          E-Mail
        </span>
        <span
          className="MuiFormLabel-asterisk-21"
        >
           *
        </span>
      </label>
      <div
        className="MuiInputBase-root-35 MuiInput-root-22 MuiInput-underline-26 MuiInputBase-formControl-36 MuiInput-formControl-23"
        data-cy="email"
        onClick={[Function]}
      >
        <input
          aria-invalid={false}
          autoComplete="email"
          autoFocus={true}
          className="MuiInputBase-input-45 MuiInput-input-30"
          disabled={false}
          id="email"
          name="email"
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          required={true}
          type="text"
          value="test@example.com"
        />
      </div>
    </div>
    <div
      className="MuiFormControl-root-4 MuiFormControl-marginNormal-5 MuiFormControl-fullWidth-7"
    >
      <label
        className="MuiFormLabel-root-15 MuiFormLabel-filled-19 MuiFormLabel-required-20 MuiInputLabel-root-8 MuiInputLabel-formControl-9 MuiInputLabel-animated-12 MuiInputLabel-shrink-11"
        data-shrink={true}
        htmlFor="password"
      >
        <span>
          Passwort
        </span>
        <span
          className="MuiFormLabel-asterisk-21"
        >
           *
        </span>
      </label>
      <div
        className="MuiInputBase-root-35 MuiInput-root-22 MuiInput-underline-26 MuiInputBase-formControl-36 MuiInput-formControl-23"
        data-cy="password"
        onClick={[Function]}
      >
        <input
          aria-invalid={false}
          autoComplete="current-password"
          className="MuiInputBase-input-45 MuiInput-input-30 MuiInputBase-inputType-48 MuiInput-inputType-33"
          disabled={false}
          id="password"
          name="password"
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          required={true}
          type="password"
          value="mypassword"
        />
      </div>
    </div>
    <button
      className="MuiButtonBase-root-78 MuiButton-root-52 MuiButton-contained-63 MuiButton-containedPrimary-64 MuiButton-raised-66 MuiButton-raisedPrimary-67 MuiButton-fullWidth-77"
      data-cy="submit"
      disabled={false}
      onBlur={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex="0"
      type="submit"
    >
      <span
        className="MuiButton-label-53"
      >
        <span>
          Anmelden
        </span>
      </span>
      <span
        className="MuiTouchRipple-root-81"
      />
    </button>
  </form>
  <a
    className="MuiButtonBase-root-78 MuiButton-root-52 MuiButton-contained-63 MuiButton-containedPrimary-64 MuiButton-raised-66 MuiButton-raisedPrimary-67 MuiButton-fullWidth-77 LoginPage-registrationButton-2"
    data-cy="registration"
    href="/register"
    onBlur={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    role="button"
    tabIndex="0"
  >
    <span
      className="MuiButton-label-53"
    >
      <span>
        Noch kein Konto? Hier registrieren!
      </span>
    </span>
    <span
      className="MuiTouchRipple-root-81"
    />
  </a>
</main>
`;
